### ISSUE TITLE
Rollup queries with count aggregator produce unexpected results

### ISSUE THREAD

**Issue by neilfordyce** [OpenTSDB/opentsdb#1683]
Came across an interesting bug in the count aggregator while setting up rollups for the first time using OpenTSDB 2.4.  

Repro as follows: 

1. Configure with rollup_config.json
```
{
    "aggregationIds": {
        "sum": 0,
        "count": 1,
        "min": 2,
        "max": 3,
        "avg": 4
    },
    "intervals": [
        {
            "table": "tsdb",
            "preAggregationTable": "tsdb",
            "interval": "1m",
            "rowSpan": "1h",
            "defaultInterval": true
        },
        {
            "table": "tsdb-rollup-1h",
            "preAggregationTable": "tsdb-rollup-1h",
            "interval": "1h",
            "rowSpan": "1d",
            "defaultInterval": false
        }
    ]
}
```

2. Post some data
```
curl -XPOST http://localhost:4242/api/rollup -d '[
  {"metric": "sys.cpu","timestamp": 1560859200,"value": 100,"tags": {"host": "1"},"interval": "1h","aggregator": "SUM"},
  {"metric": "sys.cpu","timestamp": 1560859200,"value": 50,"tags": {"host": "2"},"interval": "1h","aggregator": "SUM"},
  {"metric": "sys.cpu","timestamp": 1560862800,"value": 150,"tags": {"host": "1"},"interval": "1h","aggregator": "SUM"},
  {"metric": "sys.cpu","timestamp": 1560862800,"value": 50,"tags": {"host": "2"},"interval": "1h","aggregator": "SUM"},
  {"metric": "sys.cpu","timestamp": 1560859200,"value": 5,"tags": {"host": "1"},"interval": "1h","aggregator": "COUNT"},
  {"metric": "sys.cpu","timestamp": 1560859200,"value": 5,"tags": {"host": "2"},"interval": "1h","aggregator": "COUNT"},
  {"metric": "sys.cpu","timestamp": 1560862800,"value": 10,"tags": {"host": "1"},"interval": "1h","aggregator": "COUNT"},
  {"metric": "sys.cpu","timestamp": 1560862800,"value": 10,"tags": {"host": "2"},"interval": "1h","aggregator": "COUNT"}
]'
```

3. Run a query
```
{
    "start": 1560859200,
    "end": 1560863000,
    "showSummary": true,
    "queries": [
        {
            "downsample": "1h-count-zero",
            "aggregator": "sum",
            "metric": "sys.cpu",
            "rollupUsage": "ROLLUP_NOFALLBACK"
        }
    ]
}
```

|      Time           | Expected  | Actual  |
|---------------|-----------|--------|
|1560859200   |(5+5)=10    |    2       |
|1560862800   |(10+10)=20|      2     |

The actual results look like what I would expect from downsample:sum and aggregator:count

4. Run another query
```
{
    "start": 1560859200,
    "end": 1560863000,
    "showSummary": true,
    "queries": [
        {
            "downsample": "1h-sum-zero",
            "aggregator": "count",
            "metric": "sys.cpu",
            "rollupUsage": "ROLLUP_NOFALLBACK"
        }
    ]
}
```

|      Time           | Expected  | Actual  |
|---------------|-----------|--------|
|1560859200   | 2                | 150      |
|1560862800   | 2                | 200      |

The actual results look like what I would expect from downsample:sum and aggregator:sum

**Comment by tdinucci** [comment]
I've spent some time looking into this issue and am hoping that a maintainer of this project can help shed some light.

The documentation for the rollup feature suggests to me that rollup tables can be used to store pre-downsampled data.  My interpretation of this is that I can take all of my measurements of a specific metric across a 1h period (for example), downsample them and stick the result into the rollup table, so that I then don't have to continually downsample the larger dataset.

I'd expect that when I want to then pull back downsampled data I can specify my required downsampler, a value for subquery.rollupUsage and if there is an appropriate pre-downsampled value stored in the rollup table it'll be used rather than a downsample being calculated on the base metrics.

I'm seeing some differences between what's in the code and what I took from reading the documentation and the code is downsampling the data that I've stored in the rollup table (which I've already downsampled).

A minor change to the code resolves the issue (as I see it) however I'm not confident that I've not misunderstood the way this is meant to work.  A big reason for feeling like I'm missing something is unit tests like `TestDownsampler.testDownsampler_rollupSum`.  With my expectation of what rollup is, the way this test reads to me is; store 5s downsamples as 1h rollups and then on retreival downsample this again into 10s intervals. 

If my understanding is correct then I'm happy to submit a PR to address this.
